### ISSUE TITLE
 functional.reset_net 的使用 （Use of functional.reset_net）

### ISSUE THREAD

**Issue by HaoZhaiZi** [fangwei123456/spikingjelly#594]
基于spikingjelly的脉冲神经网络需要经过多次前向传播，才能计算损失函数。因此，请求确认一下， functional.reset_net 是否在前向传播后立即执行？（The spikingjelly based pulse neural network needs to travel forward several times to calculate the loss function. Therefore, ask to confirm that functional.reset_net is executed immediately after forward propagation?）例如：#######################################################################
            # 对输入图像 img0 和 img1 和 原始图 和视差图 分别进行前向传播
            s1, p1, d1, _ = model.forward(img0, img0_depth, flag = 0)
            # 优化一次参数后，需要重置网络的状态，因为SNN的神经元是有“记忆”的
            functional.reset_net(model)
            #######################################################################
            s2, p2, d2, _ = model.forward(img1, img1_depth, flag = 0)
            # 优化一次参数后，需要重置网络的状态，因为SNN的神经元是有“记忆”的
            functional.reset_net(model)

**Comment by Met4physics** [comment]
在一个完整的时间周期后重置

**Comment by Met4physics** [comment]
在一个完整的时间周期后重置

**Comment by HaoZhaiZi** [comment]
> 在一个完整的时间周期后重置
谢谢帮助。请问，以下例子中，网络能否正常训练？
  with amp.autocast():   # 混合精度训练
            #######################################################################
            # 对输入图像 img0 进行前向传播
            s1, p1, d1, _ = model.forward(img0, img0_depth, flag = 0)
            # 优化一次参数后，需要重置网络的状态，因为SNN的神经元是有“记忆”的
            functional.reset_net(model)
            #######################################################################
            # 对输入图像img1 分别进行前向传播
            s2, p2, d2, _ = model.forward(img1, img1_depth, flag = 0)
            # 优化一次参数后，需要重置网络的状态，因为SNN的神经元是有“记忆”的
            functional.reset_net(model)
            #######################################################################
           # 对输入图像org_img分别进行前向传播
            _, _, _, g3 = model.forward(org_img, org_img_depth, flag = 1)
            # # 优化一次参数后，需要重置网络的状态，因为SNN的神经元是有“记忆”的
            functional.reset_net(model)
            #######################################################################
            # 然后计算损失 loss 和损失项 loss_item
            loss, loss_item = model.loss(s1, p1, d1, s2, p2, d2, mat, g3, tar_img)
            # loss, loss_item = model.loss(_, _, _, _, _, _, _, g3, tar_img)

            # # 对损失进行反向传播（loss.backward()），计算模型参数的梯度
            # loss.backward()
            # 使用梯度裁剪（clip_grad_norm_）来限制模型参数梯度的范数，以防止梯度爆炸问题。
            # optim_cfg['GRAD_NORM_CLIP'] 是梯度范数裁剪的阈值
            total_norm = clip_grad_norm_(model.parameters(), optim_cfg['GRAD_NORM_CLIP'])
            # total_norm = clip_grad_value_(model.parameters(), optim_cfg['GRAD_NORM_CLIP'])
            # total_norm = 0
            # # 执行优化步骤（optimizer.step()），根据计算的梯度更新模型参数
            # optimizer.step()

        scaler.scale(loss).backward()
        scaler.step(optimizer)
        scaler.update()

**Comment by Met4physics** [comment]
看起来是可以的
